api/src/services/__tests__/competenciesService.ts

```diff
@@ -62,4 +64,28 @@ describe('competenciesService', () => {
       });
     });
   });
+
+  describe('updateCompetency', () => {
+    it('should update the specified competency if the current user is the one who originally authored the competency', async () => {
+      const id = 2;
+      const label = 'label';
+      const description = 'description';
+      mockQuery(
+        'update `competencies` set `label` = ?, `description` = ? where `id` = ?',
+        [label, description, id],
+        1
+      );
+      await updateCompetency(id, label, description);
+    });
+    it('should authorizeCompetencyUpdate for a user who authored the competency', async () => {
+      const principalId = 2;
+      const competencyId = 3;
+      mockQuery(
+        'select `id` from `competencies` where `id` = ? and `principal_id` = ?',
+        [competencyId, principalId],
+        [competencyId, principalId]
```

Comment:
The mocked response from this query doesn't appear to conform to what knex would be returning from this query. It's really important when mocking apis to ensure that the mocks behave exactly like the original apis.

Reply:
I made changes in the test matching the knex call output.